### PR TITLE
feat: support `no_std` for `ethereum-forks`

### LINKED ISSUE #8864: ethereum-forks no-std support
no-std support for https://github.com/paradigmxyz/reth/blob/e92dbe949c415f7c1ebbad812f3b905cf902cf14/crates/ethereum-forks/Cargo.toml#L9-L9

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #8864

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by emhane — APPROVED** [review]

File: crates/ethereum-forks/src/lib.rs [hunk 1]
```diff
@@ -15,6 +15,10 @@
 // TODO: remove when https://github.com/proptest-rs/proptest/pull/427 is merged
 #![allow(unknown_lints, non_local_definitions)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
```
Comment (emhane) [inline crates/ethereum-forks/src/lib.rs:21]: why not do this in the manifest?

Comment (kamuik16) [reply crates/ethereum-forks/src/lib.rs:21]: I have used this as a reference from earlier PRs like https://github.com/paradigmxyz/reth/pull/8729 and https://github.com/paradigmxyz/reth/pull/8817.

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**